internal/metrics/source/prometheus/convert.go

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) F5, Inc.
```

Comment:
That would ideally be the case, but I'm not sure that's true? For example a Prometheus histogram is represented as a series of datapoints where:
* Datapoints whose name is suffixed with `_bucket` represent a 'bucket' data point
* Datapoints whose name is suffixed with `_sum` represent a 'sum' data point
* Datapoints whose name is suffixed with `_count` represent a 'count' data point

`convert.go` assumes the above for its parsing. 

Will we force the other data sources to transform their histograms to conform to the model above? If yes, then I would agree that we move `convert.go` to `internal/metrics`, otherwise I would say each exporter will need its own `convert.go`.

Reply:
Then there's the issue of how the temporality of metrics for the metrics source. Different sources may use different temporality, which means we would need to at least be able to somehow parameterize that in the converter.

I would be inclined to leave this as is for now and then look at what a generic converter would look like when we add our next source of metrics.

Reply:
We discussed this at the office and decided that we'll keep it as is for now and then refactor to a common converter if it makes sense when we add the next data source.